include/rdma/fi_ext.h

```diff
@@ -43,22 +43,25 @@ extern "C" {
 #endif
 
 /*
- * Each provider can define provider-specific values by
- * choosing an unique 11-bit (since the most significant
- * bit must be 1) provider-specific code to avoid
- * overlapping with other providers. E.g.,
+ * Each provider needs to define an unique 12-bit provider
+ * specific code to avoid overlapping with other providers,
+ * then bit left shift the code 16 bits. The low 16 bits
+ * can be used to define provider specific values.
+ * E.g.,
+ *
+ * define FI_PROV_SPECIFIC_XXX    (0xabc << 16)
  *
- * #define FI_PROV_SPECIFIC_XXX    (FI_PROV_SPECIFIC | 0xabc << 20)
  * enum {
- *        FI_PROV_XXX_VALUE        (1U | FI_PROV_SPECIFIC_XXX)
+ *        FI_PROV_XXX_FOO = -(FI_PROV_SPECIFIC_XXX),
+ *        FI_PROV_XXX_BAR,
  * }
  */
 
-#define FI_PROV_SPECIFIC_EFA   (FI_PROV_SPECIFIC | 0xefa << 20)
+#define FI_PROV_SPECIFIC_EFA   (0xefa << 16)
 
-/* EFA options */
-enum {
-       FI_OPT_EFA_RNR_RETRY = (1U | FI_PROV_SPECIFIC_EFA),
+/* negative options are provider specific */
+enum fi_efa_opt {
```

Comment:
Why did we name the enum?

Reply:
Oh, that's a feedback I got in the initial change in PR #7065 , which was posted after that PR was merged. 
I just saw the feedback, so changed it here.

Reply:
I would remove the name.  "enum fi_efa_opt" will not be used anywhere, particularly in the public API, so it's only serving as a comment.